### DISCUSSION TITLE
Vector VN16XX Device Connection

### DISCUSSION THREAD

**Discussion by DrewBallins** [collin80/SavvyCAN#739] [Q&A]
Has anybody used this software with a Vector VN16XX device before (such as a VN1640)? I have the latest Vector device drivers installed but am having trouble creating a connection to my VN1640 device. I'm not sure what to put in the 'Port' field in SavvyCAN as the list is not auto-populated and I'm not seeing any specific help on this, should that be '0'? '1'? 'can0' (this is what would make the most sense to me based on the Qt docs [here](https://doc.qt.io/qt-6/qtserialbus-vectorcan-overview.html) but still doesn't work)? I've tried many options and all result in the Device having a status of 'Not Connected' and not being able to send or receive CAN traffic via the VN1640. Is there any additional setup needed in Vector Hardware Config? Any help would be greatly appreciated!
<img width="542" alt="vectorcan Connection Settings" src="https://github.com/collin80/SavvyCAN/assets/57241202/35434a03-220f-47fb-974f-e596ceaf90bb">

**Comment by jerinj89** [comment]
@DrewBallins: Could you resolve the issue?

**Comment by yoimiya52** [comment]
Maybe I can provide some ideas, after my observation, QCanBus is not able to complete the setting of two key reserved fields and canfdkey when setting on canfd. I didn't find a solution for this, it may be a bug in QCanBus itself; in my scenario, I abandoned QCanBus and directly used the driver provided by Vector to connect to the VN device, which enabled the capture of can as well as canfd traffic.

**Comment by phophollety** [comment]
Hello, CAN I help? I've VN1630 and VN5610, I'd love to send some reports or whatever

**Reply by macskyy** [reply]
> Hello, CAN I help? I've VN1630 and VN5610, I'd love to send some reports or whatever

I have an issue when reading data from an device, the ID always shown 0x000. In the actual, ID device is 15. Could you help me to debug this problem? thank you.

**Reply by KrisK63** [reply]
Yes, you can. Please provide a hex dumping of your vn1630.
Also,  what piggies are you running?
